packages/providers/test/retry.test.ts

```diff
@@ -52,7 +52,7 @@ describe('Retries correctly', () => {
     // this might be something related to the event loop and how it schedules setTimeouts
     // expectedWaitTime minus 5ms seems like reasonable to allow
     expect(waitTime).toBeGreaterThanOrEqual(expectedWaitTime - 5);
-    expect(waitTime).toBeLessThanOrEqual(expectedWaitTime + 15);
+    expect(waitTime).toBeLessThanOrEqual(expectedWaitTime + 50);
```

Comment:
Isn't this still prone to flakiness?

Timeouts and small ms-drifts are fragile, as things can always run faster/slower.

Reply:
I agree and that's the reason why we were having flakiness, but I don't have an idea how to test it in another manner. I'd be happy to discuss another take on it. We can also leave it with 50ms and see if the flakiness returns and then consider it further.